Sources/XcodeProj/Project/XcodeProj.swift

```diff
@@ -133,74 +148,122 @@ extension XcodeProj: Writable {
         path + "xcshareddata"
     }
 
-    /// Returns schemes folder path relative to the given path.
+    /// Writes shared data to the given path.
+    ///
+    /// - Parameter path: path to `.xcodeproj` file.
+    /// - Parameter override: if shared data should be overridden. Default is true.
+    /// - Parameter outputSettings: Controls the writing of various files.
+    ///   If false will throw error if shared data already exists at the given path.
+    public func writeSharedData(path: Path, override: Bool = true) throws {
+        try sharedData?.write(path: XcodeProj.sharedDataPath(path), override: override)
+    }
+
+    /// Returns user data path relative to the given path.
     ///
     /// - Parameter path: `.xcodeproj` file path
+    /// - Returns: user data path relative to the given path.
+    public static func userDataPath(_ path: Path) -> Path {
+        path + "xcuserdata"
+    }
+
+    /// Returns user data path for a specific user relative to the given path.
+    ///
+    /// - Parameter path: `.xcodeproj` file path
+    /// - Returns: user data path relative to the given path.
+    public static func userDataPath(_ path: Path, userName: String) -> Path {
+        XcodeProj.userDataPath(path) + "\(userName).xcuserdatad"
+    }
+
+    /// Writes user data to the given path.
+    ///
+    /// - Parameter path: path to `.xcodeproj` file.
+    /// - Parameter override: if user data should be overridden. Default is true.
+    /// - Parameter outputSettings: Controls the writing of various files.
+    ///   If false will throw error if user data already exists at the given path.
+    public func writeUserData(path: Path, override: Bool = true) throws {
+        let userDataPath = XcodeProj.userDataPath(path)
+        if override, userDataPath.exists {
+            try userDataPath.delete()
+        }
+
+        try userDataPath.mkpath()
+        for userData in userData {
+            try userData.write(path: XcodeProj.userDataPath(path, userName: userData.userName), override: true)
+        }
+    }
+
+    /// Returns schemes folder path relative to the given path.
+    ///
+    /// - Parameter path: parent folder of schemes folder (xcshareddata or xcuserdata)
     /// - Returns: schemes folder path relative to the given path.
     public static func schemesPath(_ path: Path) -> Path {
```

Comment:
These methods now need the `xcshareddata` or `xcuserdata` path instead of the `.xcodeproj` path because they need to it to produce a relative path to that folder instead of a path always relative to the `xcshareddata` folder. 

```
public static func schemesPath(path:)
public static func schemePath(path:schemeName:)
public static func debuggerPath(path:)
public static func breakPointsPath(path:)
public static func schemeManagementPath(path:)            <-- New
```

The same goes for the writing methods, they need a path to the `xcshareddata` or `xcuserdata` folder instead of the `.xcodeproj` path. 

```
public static func writeSchemes(schemes:path:override:)
public static func writeBreakPoints(breakpoints:path:override:)
```